docs/cppreference.rst

```diff
@@ -0,0 +1,49 @@
+Automatic cross-linking to cppreference.com
+===========================================
+
+This theme provides an optional :python:`sphinx_immaterial.cppreference`
+extension that provides automatic linking to C/C++ standard symbols documented
+at https://cppreference.com.
+
+To use this extension, simply add it to the list of extensions in :file:`conf.py`:
+
+.. code-block:: python
+
+    extensions = [
+        # other extensions...
+        "sphinx_immaterial.cppreferences",
+    ]
+
+.. rst-example::
+
+   .. cpp:function:: template <typename T> \
+                     std::enable_if_t<std::is_same_v<T, char> || \
+                                      std::is_same_v<T, uint8_t>, \
+                                      std::string> \
+                     ConvertVectorToString(std::vector<T> x);
+
+      Converts an :cpp:expr:`std::vector` to an :cpp:expr:`std::string`.
+
+.. confval:: cppreference_xml_files
```

Comment:
This option seems like it would be required if using the ext in another theme that doesn't ship with the XML files used as a default here. Does the extension somehow use this theme's xml files (even for other themes)?

Reply:
The files are just data files of the package, and they are just accessed using the normal `importlib.resources` machinery, so they will work fine even if not using this theme.  They are only used during the documentation generation process, but they don't get included in the output as static files.